pkg/gateway/api/api_grpc.go

```diff
@@ -864,7 +864,7 @@ func (s *grpcGatewayService) GetSegmentUsers(
 	updatedSegments := make([]*featureproto.SegmentUsers, 0, len(targetSegmentUsers))
 	adjustedRequestedAt := req.RequestedAt - secondsForAdjustment
 	for _, su := range targetSegmentUsers {
-		if su.UpdatedAt < adjustedRequestedAt {
+		if su.UpdatedAt > adjustedRequestedAt {
```

Comment:
The `UpdatedAt` must be higher than the `requestedAt`, so I can return only the updated segments.